numpy/core/setup.py

```diff
@@ -496,7 +496,7 @@ def generate_config_h(ext, build_dir):
                 # add the guard to make sure config.h is never included directly,
                 # but always through npy_config.h
                 target_f.write(textwrap.dedent("""
-                    #ifndef _NPY_NPY_CONFIG_H_
+                    #ifndef NUMPY_CORE_SRC_COMMON_NPY_CONFIG_H_
```

Comment:
This never worked correctly as it allows `config.h` to be included anywhere after `npy_config.h` as well as in `npy_config.h` itself.